complete.go

```diff
@@ -46,7 +47,12 @@ func completerNg(line string, pos int) (prefix string, completions []string, suf
 			res = append(res, scope.VarName)
 		}
 	}
-	// TODO add type matches ? (e.g. float64, int, ...)
+	// TODO add user-types completion.
```

Comment:
If you export cur in typecheck.Checker, you can walk that scope just as you do above for the evaluator scope, to get both the user-defined types and the universal types.